### ISSUE TITLE
request: mapping in geom_* and stat_*

### ISSUE THREAD

**Issue by davidswelt** [yhat/ggplot#506]
The geom_ and stat_ functions should support specific mapping arguments.
For example, I don't see how I can make a plot that shows color-coded points as well as a single regression or trend line (stat_smooth) (rather than one for each color).

**Comment by schmeing** [comment]
I face the same restrictions. Making the data= parameter work would also help. Currently it seems impossible to overlay multiple plots.

**Comment by schmeing** [comment]
I made myself a workaround:
Added this definition to geom.py:
```py
    def _update_data(self, data, _aes):
        if 'mapping' in self.params:
            _aes = self.params['mapping']
        if 'data' in self.params:
            data = self.params['data']
        return (data, _aes)
```
and then at the beginning of `def plot(self, ax, data, _aes):` for geometry of choice (in my case geom_line.py):
```py
(data, _aes) = self._update_data(data, _aes)
```

I have no idea what that might break or not, because I didn't really understand some of the code:
For example why in in `def _get_plot_args(self, data, _aes):` the data is first put into mpl_params just to be removed from it again later

**Comment by glamp** [comment]
This just *might* work. I will run the tests and see what happens!

@schmeing can you submit a PR?

**Comment by schmeing** [comment]
Did break the aes evaluations like 'x+1'
Pushed a fix

**Comment by glamp** [comment]
merged and on master

**Comment by glamp** [comment]
released in 0.11
